VideoLocker/src/main/java/org/edx/mobile/util/LocaleUtils.java

```diff
@@ -0,0 +1,42 @@
+package org.edx.mobile.util;
+
+import android.support.annotation.NonNull;
+
+import java.util.Locale;
+
+public class LocaleUtils {
+
+    public static String getCountryNameFromCode(@NonNull String countryCode) throws InvalidLocaleException {
+        final String uppercaseRegion = countryCode.toUpperCase(Locale.ROOT);
+        if (!isValidBcp47Alpha(uppercaseRegion, 2, 2)) {
```

Comment:
Yeah, it seems that the builder requires either a valid BCP 47 Alpha or UN M.49 for the area code. Maybe we should support it as well?